applications/ShapeOptimizationApplication/test_examples/06_Eigenfrequency_Gradient_Verification/run_verification.py

```diff
@@ -0,0 +1,92 @@
+# Import Kratos core and apps
+from KratosMultiphysics import *
+from KratosMultiphysics.StructuralMechanicsApplication import *
+import structural_response_function_factory
+
+# ==============================================================================
+# Define parameters
+kratos_response_settings = Parameters("""
+{
+            "response_type"          : "eigenfrequency",
+            "primal_settings"        : "parameters.json",
+            "gradient_mode"          : "semi_analytic",
+            "step_size"              : 1e-8,
+            "traced_eigenfrequencies": [1]
+}""")
+
+# Note that depending on move direction the initial pertubations might be too big leading to a crashing simulation
+move_node_id = 28
+move_direction = 'z'
+levels_of_finite_differencing = 12
+
+# ==============================================================================
+# Some preprocessing
+with open(kratos_response_settings["primal_settings"].GetString(),'r') as parameters:
+    parameters = Parameters(parameters.read())
+model_part_name = parameters["solver_settings"]["model_part_name"].GetString()
+
+# Compute reference values
+model = Model()
+response = structural_response_function_factory.CreateResponseFunction(kratos_response_settings["response_type"].GetString(), kratos_response_settings, model)
+response.RunCalculation(True)
+response_values = [response.GetValue()]
```

Comment:
i would directly store it as `reference_value` and not in this list, then you also don't need the `"-"` in the deltas list below...